src/Instrument/ClassLoading/CacheWarmer.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/*
+ * Go! AOP framework
+ *
+ * @copyright Copyright 2013, Lisachenko Alexander <lisachenko.it@gmail.com>
+ *
+ * This source file is subject to the license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Go\Instrument\ClassLoading;
+
+use Go\Core\AspectKernel;
+use Go\Instrument\FileSystem\Enumerator;
+use Go\Instrument\Transformer\FilterInjectorTransformer;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Warms up the cache
+ */
+class CacheWarmer
+{
+    /**
+     * @var AspectKernel
+     */
+    protected $aspectKernel;
+
+    /**
+     * @var OutputInterface
+     */
+    protected $output;
+
+    /**
+     * CacheWarmer constructor.
+     *
+     * @param AspectKernel $aspectKernel A kernel
+     * @param OutputInterface $output Optional output to log current status.
+     */
+    public function __construct(AspectKernel $aspectKernel, OutputInterface $output = null)
+    {
+        $this->aspectKernel = $aspectKernel;
+        $this->output = $output;
```

Comment:
What I don't like here is that cache warmer depends on OutputInterface.

Kinda seams natural that it should depend on LoggerInterface instead. Warmer is being used for console command, true, but it is used in warming process, per example, in Symfony.

Things to consider

- If we use LoggerInterface instead of OutputInterface, we will use nice output formatting in Console.
- Warmer on the other hand should log its output in warming process, especially in dev mode.

These are my thoughts on this subject, but I don't have any elegant solution for it....